api/api.go

```diff
@@ -0,0 +1,267 @@
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (api *ImportAPI) getDimensions(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	instanceID := vars["instanceId"]
+	dimensions, dataStoreError := api.dataStore.GetDimension(instanceID)
+	if dataStoreError != nil {
+		setErrorCode(w, dataStoreError)
+		return
+	}
+	bytes, error := json.Marshal(dimensions)
+	if error != nil {
+		log.Error(error, log.Data{"instance_id": instanceID, "dimensions": dimensions})
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+	setJSONContentType(w)
+	_, writeError := w.Write(bytes)
+	if writeError != nil {
+		log.Error(error, log.Data{"instance_id": instanceID, "dimensions": dimensions})
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+	log.Info("returned dimensions", log.Data{"instance_id": instanceID})
+}
+
+func (api *ImportAPI) addDimension(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	instanceID := vars["instanceId"]
+	dimensionName := vars["dimension_name"]
+	dimensionValue := vars["value"]
+	if dimensionName == "" || dimensionValue == "" {
+		log.Error(fmt.Errorf("Missing parameters"), log.Data{"instance_id": instanceID, "name": dimensionName, "value": dimensionValue})
+		http.Error(w, "Bad client request received", http.StatusBadRequest)
+		return
+	}
+	dimension := models.Dimension{Name: dimensionName, Value: dimensionValue}
+	dataStoreError := api.dataStore.AddDimension(instanceID, &dimension)
+	if dataStoreError != nil {
+		setErrorCode(w, dataStoreError)
+		return
+	}
+	log.Info("dimension added", log.Data{"instance_id": instanceID, "name": dimensionName, "value": dimensionValue})
+}
+
+func (api *ImportAPI) addNodeID(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	instanceID := vars["instanceId"]
+	dimensionName := vars["dimension_name"]
+	nodeValue := vars["value"]
+	if dimensionName == "" || nodeValue == "" {
+		log.Error(fmt.Errorf("Missing parameters"), log.Data{"instance_id": instanceID, "name": dimensionName, "nodeValue": nodeValue})
+		http.Error(w, "Bad client request received", http.StatusBadRequest)
+		return
+	}
+	dimension := models.Dimension{NodeID: nodeValue}
+	dataStoreError := api.dataStore.AddNodeID(instanceID, dimensionName, &dimension)
+	if dataStoreError != nil {
+		log.Error(dataStoreError, log.Data{"instance_id": instanceID, "name": dimensionName, "nodeValue": nodeValue})
+		setErrorCode(w, dataStoreError)
+		return
+	}
+	log.Info("added node id", log.Data{"instance_id": instanceID, "name": dimensionName, "nodeValue": nodeValue})
+}
+
+func setErrorCode(w http.ResponseWriter, err error) {
+	switch {
+	case err == errors.JobNotFoundError:
+		http.Error(w, "Import job not found", http.StatusNotFound)
+		return
+	case err != nil:
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+}
+
+func setJSONContentType(w http.ResponseWriter) {
```

Comment:
Unsure if these needs to be in its own function as it is just one line of code anyway